homeassistant/components/modbus/cover.py

```diff
@@ -89,22 +81,23 @@ def __init__(
         # If we read cover status from coil, and not from optional status register,
         # we interpret boolean value False as closed cover, and value True as open cover.
         # Intermediate states are not supported in such a setup.
-        if self._coil is not None:
+        if self._input_type == CALL_TYPE_COIL:
             self._write_type = CALL_TYPE_WRITE_COIL
+            self._write_address = self._address
             if self._status_register is None:
                 self._state_closed = False
                 self._state_open = True
                 self._state_closing = None
                 self._state_opening = None
-
-        # If we read cover status from the main register (i.e., an optional
-        # status register is not specified), we need to make sure the register_type
-        # is set to "holding".
-        if self._register is not None:
+        else:
+            # If we read cover status from the main register (i.e., an optional
+            # status register is not specified), we need to make sure the register_type
+            # is set to "holding".
```

Comment:
Where is this comment applied?

Reply:
Good catch, forgot to remove it, while moving the code around. Removed.